python/ray/data/dataset.py

```diff
@@ -2830,21 +2832,43 @@ def to_tf(
     ) -> "tf.data.Dataset":
         """Return a TF Dataset over this dataset.
 
-        The TF Dataset will be created from the generator returned by the
-        ``iter_batches`` method. ``prefetch_blocks`` and ``batch_size``
-        arguments will be passed to that method.
-
-        This is only supported for datasets convertible to Arrow records.
-
-        It is recommended to call ``.split()`` on this dataset if
-        there are to be multiple TensorFlow workers consuming the data.
+        .. tip::
+            If multiple TensorFlow workers consume this data, call
+            :meth:`~ray.data.Dataset.split`.
 
         .. warning::
-
-            If your dataset contains ragged tensors, this method will error. To prevent
+            If your dataset contains ragged tensors, this method errors. To prevent
             errors, resize tensors or
             :ref:`disable tensor extension casting <disable_tensor_extension_casting>`.
```

Comment:
The only blocker for this is that schema inference breaks for ragged tensors; IMO we should provide an optional `output_signature` argument that bypasses schema inference (I think this is a generically good thing to have), at which point ragged tensors would currently be supported via specifying an `output_signature` with the appropriate ragged tensor spec.

>disable tensor extension casting

For `.to_tf()`, I don't think that ragged tensors will work, even with tensor extension casting disabled. I think the only route is resizing the tensors.

Reply:
https://github.com/ray-project/ray/pull/29476 adds support for ragged tensors and removes this warning.